### DISCUSSION TITLE
Support for magazines and epapers in PDF format?

### DISCUSSION THREAD

**Discussion by tomillr** [noDRM/DeDRM_tools#578] [Q&A]
Is it possible to remove DRM from magazines in PDF format as well? Or ist DeDRM intended only to work with regular ebooks?

I've tried it with a PDF magazine in calibre, but the end result is a completely revamped version with different fonts and a changed page layout. Is there a way to disable the DRM without changing the documents layout?

**Comment by ElleKayEm** [comment]
Is this in calibre's ebook viewer?  Try an actual pdf viewer.  Even different pdf viewers can display them very differently.  DeDRM just decrypts, it doesn't change anything.

**Reply by tomillr** [reply]
Yes,I've tried different pdf viewers on MacOs and Windows. I can replicate this issue with every PDF by this one publication. All other ten or so publications I've tried work fine no matter the pdf viewer or OS.

I've made some screenshots to illustrate the problem:

The PDF with an active license works fine in Adobe Digital Edition:
<img width="1292" alt="Bildschirmfoto 2024-07-29 um 10 17 30" src="https://github.com/user-attachments/assets/419b10d2-62d6-426e-a535-f0273c524c0a">

When I drag this file into calibre and open the resulting PDF from the calibre library with a PDF viewer, the problems begin.
<img width="1477" alt="Bildschirmfoto 2024-07-29 um 10 18 47" src="https://github.com/user-attachments/assets/05fc2912-eaab-4c68-922f-f176ef2e121e">

The native PDF viewer in MacOS show the pages in a different order. For example, page 7 in the PDF is actually displayed as page 35 from the magazine:

<img width="1269" alt="Bildschirmfoto 2024-07-29 um 10 20 10" src="https://github.com/user-attachments/assets/a226e131-ed0a-4c69-ab1a-99576785c14a">

The pages also become blank after the first 25 to 35 pages:

<img width="1269" alt="Bildschirmfoto 2024-07-29 um 10 19 45" src="https://github.com/user-attachments/assets/8db5e5f4-c65a-4560-b493-835dd6ff8f05">

When I try to open the PDF in Adobe Acrobat on Windows, all pages are empty and I get an error message.

![Screenshot Adobe Reader](https://github.com/user-attachments/assets/8519edc9-6918-4779-b8ed-328399f3a233)

**Comment by tomillr** [comment]
I've done some more testing. Apparently, this affects all issues of a particular German tech publication. Other PDF magazines and papers work fine. The PDF works like intended when viewed in Adobe Digital Edition. But once I've pulled the PDF into calibre, the page order ist messed up (it starts at the back and goes backwards) and after 20 pages or so it show just empty, white pages.

**Reply by ElleKayEm** [reply]
Not a DeDRM issue.

**Reply by tomillr** [reply]
I don't want to argue, but how is this not an DeDRM issue? The encrypted PDF is fine. When I drag the file into calibre without DeDRM installed, the PDF is just a collection of empty pages, naturally. Once I install the DeDRM plug-in, the cover as well as 25 to 30 random pages are displayed. After that, the rest of the 100 or so pages is encrypted (blank).

**Reply by ElleKayEm** [reply]
You didn't see my previous comment?
